azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.keyvault.spring;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.util.ClassUtils;
+
+public class KeyVaultEnvironmentPostProcessor implements EnvironmentPostProcessor {
+
+    @Override
+    public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+
+        if (isKeyVaultEnabled(environment)) {
+            final KeyVaultEnvironmentPostProcessorHelper helper =
+                    new KeyVaultEnvironmentPostProcessorHelper(environment);
+            helper.addKeyVaultPropertySource();
+        }
+    }
+
+    private boolean isKeyVaultEnabled(ConfigurableEnvironment environment){
+        if (environment.getProperty(Constants.AZURE_CLIENTID) == null) {
+            // User doesn't want to enable Key Vault property initializer.
+            return false;
+        }
+
+        boolean enabled = true;
+        if (environment.getProperty(Constants.AZURE_KEYVAULT_ENABLED) != null) {
+            enabled = Boolean.parseBoolean(environment.getProperty(Constants.AZURE_KEYVAULT_ENABLED));
+        }
+
+        return enabled && isKeyVaultClientAvailable();
+    }
+
+    private boolean isKeyVaultClientAvailable(){
+        return ClassUtils.isPresent("com.microsoft.azure.keyvault.KeyVaultClient", null);
```

Comment:
@snicoll Changing null to `ClassLoader.getSystemClassLoader()` doesn't work.

Reply:
I see null is used a lot in spring boot source code for this `ClassUtils.isPresent` call.

If changing to `ClassLoader.getSystemClassLoader()`,

1. It works well if directly run in Intellij.
2. doesn't work if `java -jar *.jar`

Reply:
I didn't refer to the system classloader. I referred to the `Class`'s `ClassLoader`. So `KeyvaultEnvironmentPostProcessor.class.getClassLoader()`